tests/core/test_hook_function.py

```diff
@@ -1,16 +1,30 @@
 from argparse import Namespace
-from functools import partial
+from functools import partial, reduce
 from typing import Dict
 
+import pytest
 import torch
 import torch.nn as nn
 from transformers import AutoModelForCausalLM, AutoTokenizer
 
 from flex_model.core import FlexModel, HookFunction
+from tests.fixtures import opt_350m, opt_tokenizer
 
 # could be any MLP layer and the code won't break. The test doesn't generalize
 # to other kinds of layers
-MODULE_NAME = "model.layers.27.mlp"
+MODULE_NAME = "model.decoder.layers.9.fc2"
+
+
+def rgetattr(module, attr):
+    def _getattr(module, attr):
+        return getattr(module, attr)
+
+    return reduce(_getattr, [module] + attr.split("."))
```

Comment:
Why can't we just call `getattr` directly? What's the use of a private function here?

Reply:
`getattr` and `setattr` unfortunately don't work recursively, ie. to get the module name.